Draft for the version management of the project (#132)
developersIndia/website: .github/CONTRIBUTING.md

So, with that concludes the section on "_How to contribute to the project?_". If you feel the section lacks something, please open an [Issue](https://github.com/developersIndia/website/issues/new/choose)/[Discussion](https://github.com/developersIndia/website/discussions) thread along with a referenced PR with the changes & updates.

## Managing Version Release for the Project

> Versioning a software project isn't a standard practice for no reason. It helps manage development goals in both, the long-run & the short-run. And if things goes south because of a bug or a breaking change, we can revert back to the previous working version of the project with much ease! So, as you can versioning the project is a necessity.

Reviewer: ```suggestion
Versioning a software project isn't standard practice for no reason. It helps manage development goals in both, the long run & the short run. And if anything goes south because of a bug or a breaking change, we can revert to the previous working version of the project with much ease! Therefore versioning the project is a necessity.
```

> And this section of the document details how we manage version for the project. Do note, backwards compatability isn't a requirement for web-based projects such this current one, hence SemVer is obviously not applicable to it. We did come up with a more viable option for versioning the project after countless discussion & brainstorming session. But keep in mind, anything written here isn't set in stone. So, if you've a better version management idea, then please feel free to share it with us.

Reviewer: ```suggestion
And this section of the document details how we manage the version for the project. Do note, backward compatibility isn't a requirement for web-based projects such as this one, hence `SemVer` does not apply here. We did come up with a more viable option for versioning the project. But keep in mind, anything written here isn't set in stone. So, if you have got a better version management idea, then please feel free to share it with us.
```

That said, let's dive in to the intricate details of how we manage the project's versions.

### How Often Do We Release a New Version

The nature of the project makes it difficult to maintain a fixed release schedule, hence, it's suggested to instead release a new version (_or tag_) when there are "_important feature updates_" to the project. The said "_important feature updates_" could be say;

1. Developing the Hero section of the landing page.
2. Adding a "_info_" section with information about the community.
3. A new "_About Us_" tab in the navigation bar

....and so on.

Breaking down the [end goal (_figma template_)](https://www.figma.com/file/jxpIq8GzilqzgyLpFBySP2/Developers-India-%3C3?node-id=5%3A58734) into smaller components which should be developed individually will help manage public contributions more easily!

So, upon completion of each component, it would be advisable to perform a release. More about how each releases should be versioned & the procedure for releasing is detailed in the next few sections.

### How Do We Version the Project

> The project in question is a website & as such backwards compatibility isn't a requirement. Hence, straight off-the-bat, releasing versions of the project isn't possible using [SemVer](https://semver.org/). So, its recommended to follow a different approach for versioning specific releases of the project. Taking quite a lot of inspiration from [this comment](https://github.com/developersIndia/website/discussions/69#discussioncomment-2130963) & the parent thread, we follow a "_Year-Month_" versioning system instead.

Reviewer: Let's remove unnecessary stuff here. We need to get to the point since we already discussed semver is not an option earlier

```suggestion
Taking quite a lot of inspiration from [this comment](https://github.com/developersIndia/website/discussions/69#discussioncomment-2130963) & the parent thread, we follow a "**Year-Month**" versioning system instead.
```
